File: app/src/main/java/com/pandatone/kumiwake/member/AddMember.java
package com.pandatone.kumiwake.member;

import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.Rect;
import android.os.Bundle;
import android.support.design.widget.TextInputLayout;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.AppCompatEditText;
import android.text.InputFilter;
import android.text.Spanned;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.view.ViewTreeObserver;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Toast;

import com.pandatone.kumiwake.R;
import com.pandatone.kumiwake.adapter.MemberListAdapter;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by atsushi_2 on 2016/02/24.
 */
public class AddMember extends AppCompatActivity {


    private static AppCompatEditText nameEditText;
    private static AppCompatEditText readEditText;
    private static TextInputLayout textInputLayout;
    private static RadioGroup sexGroup;
    private static RadioButton sexButton;
    private static EditText ageEditText;
    private static EditText gradeEditText;
    private static Button belongSpinner;
    private static Button roleSpinner;
    private static MemberListAdapter dbAdapter;
    private String[] beforeBelong, afterBelong;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.add_member);
        ButterKnife.bind(this);
        dbAdapter = new MemberListAdapter(this);
        beforeBelong = null;
        afterBelong = null;
        findViews();
        ageEditText.setText("");
        gradeEditText.setText("");
        Intent i = getIntent();
        int position = i.getIntExtra("POSITION", -1);
        if (position != -1) {
            setItem(position);
        }
    }

    public void findViews() {
        nameEditText = (AppCompatEditText) findViewById(R.id.input_name);
        readEditText = (AppCompatEditText) findViewById(R.id.input_name_read);
        InputFilter inputFilter = new InputFilter() {
            @Override
            public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
                if (source.toString().matches("^[a-zA-Z0-9ぁ-ん]+$")) {
                    return source;
                }
                return "";
            }
        };
        InputFilter[] filters = {inputFilter};
        readEditText.setFilters(filters);
        textInputLayout = (TextInputLayout) findViewById(R.id.member_form_input_layout);
        sexGroup = (RadioGroup) findViewById(R.id.sexGroup);
        ageEditText = (EditText) findViewById(R.id.input_age);
        gradeEditText = (EditText) findViewById(R.id.input_grade);
        belongSpinner = (Button) findViewById(R.id.select_group_spinner);
        roleSpinner = (Button) findViewById(R.id.select_role_spinner);
    }

    public void setItem(int position) {
        Name listItem = FragmentMember.nameList.get(position);
        int listId = listItem.getId();
        String name = listItem.getName();
        String name_read = listItem.getName_read();
        String sex = listItem.getSex();
        String age = String.valueOf(listItem.getAge());
        String grade = String.valueOf(listItem.getGrade());
        String belong = MemberClick.ViewBelong(position);
        String role = listItem.getRole();

        nameEditText.setText(name);
        readEditText.setText(name_read);
        if (sex != null && sex.equals("女")) {
            sexGroup.check(R.id.womanBtn);
        } else {
            sexGroup.check(R.id.manBtn);
        }
        ageEditText.setText(age);
        gradeEditText.setText(grade);
        belongSpinner.setText(belong);
        roleSpinner.setText(role);

        update(listId);

    }


    @OnClick(R.id.select_group_spinner)
    void onSelectGroupSpinnerClicked(final View view) {
        // 選択中の候補を取得
        final String buttonText = belongSpinner.getText().toString();
        String[] textArray = buttonText.split(",");
        // 候補リスト
        List<String> list = new ArrayList<String>();
        for (int j = 0; j < FragmentGroup.ListCount; j++) {
            GroupListAdapter.Group listItem = FragmentGroup.nameList.get(j);
            String groupName = listItem.getGroup();

            list.add(groupName);
        }
        final String[] belongArray = list.toArray(new String[0]);
        // 選択リスト
        boolean[] checkArray = new boolean[belongArray.length];
        for (int i = 0; i < belongArray.length; i++) {
            checkArray[i] = false;
            for (String data : textArray) {
                if (belongArray[i].equals(data)) {
                    checkArray[i] = true;
                    break;
                }
            }
        }
        // ダイアログを生成
        AlertDialog.Builder dialog = new AlertDialog.Builder(this);
        // 選択イベント
        dialog.setMultiChoiceItems(belongArray, checkArray, new DialogInterface.OnMultiChoiceClickListener() {
            @Override
            public void onClick(DialogInterface dia, int value, boolean isChecked) {
                String text = belongSpinner.getText().toString();
                // 選択された場合
                if (isChecked) {
                    // ボタンの表示に追加
                    belongSpinner.setText(text + ("".equals(text) ? "" : ",") + belongArray[value]);
                } else {
                    // ボタンの表示から削除
                    if (text.indexOf(belongArray[value] + ",") >= 0) {
                        belongSpinner.setText(text.replace(belongArray[value] + ",", ""));
                    } else if (text.indexOf("," + belongArray[value]) >= 0) {
                        belongSpinner.setText(text.replace("," + belongArray[value], ""));
                    } else {
                        belongSpinner.setText(text.replace(belongArray[value], ""));
                    }
                }
            }
        });
        dialog.setPositiveButton(getText(R.string.decide), null);
        dialog.setNeutralButton(getText(R.string.clear), new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int value) {
                belongSpinner.setText("");
                // 再表示
                onSelectGroupSpinnerClicked(view);
            }
        });
        dialog.setNegativeButton(getText(R.string.cancel), new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int value) {
                // 選択前の状態に戻す
                belongSpinner.setText(buttonText);
            }
        });
        dialog.show();

        beforeBelong = textArray;;
    }

    @OnClick(R.id.select_role_spinner)
    void onSelectRoleSpinnerClicked(final View view) {
        // 選択中の候補を取得
        final String buttonText = roleSpinner.getText().toString();
        String[] textArray = buttonText.split(",");

        // 候補リスト
        String[] strs = getResources().getStringArray(R.array.role);
        List<String> list = new ArrayList<String>(Arrays.asList(strs)); // 新インスタンスを生成
        list.remove(0);
        final String[] likeArray = list.toArray(new String[list.size()]);

        // 選択リスト
        boolean[] checkArray = new boolean[likeArray.length];
        for (int i = 0; i < likeArray.length; i++) {
            checkArray[i] = false;
            for (String data : textArray) {
                if (likeArray[i].equals(data)) {
                    checkArray[i] = true;
                    break;
                }
            }
        }
        // ダイアログを生成
        AlertDialog.Builder dialog = new AlertDialog.Builder(this);
        // 選択イベント
        dialog.setMultiChoiceItems(likeArray, checkArray, new DialogInterface.OnMultiChoiceClickListener() {
            @Override
            public void onClick(DialogInterface dia, int value, boolean isChecked) {
                String text = roleSpinner.getText().toString();
                // 選択された場合
                if (isChecked) {
                    // ボタンの表示に追加
                    roleSpinner.setText(text + ("".equals(text) ? "" : ",") + likeArray[value]);
                } else {
                    // ボタンの表示から削除
                    if (text.indexOf(likeArray[value] + ",") >= 0) {
                        roleSpinner.setText(text.replace(likeArray[value] + ",", ""));
                    } else if (text.indexOf("," + likeArray[value]) >= 0) {
                        roleSpinner.setText(text.replace("," + likeArray[value], ""));
                    } else {
                        roleSpinner.setText(text.replace(likeArray[value], ""));
                    }
                }
            }
        });
        dialog.setPositiveButton(getText(R.string.decide), null);
        dialog.setNeutralButton(getText(R.string.clear), new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int value) {
                roleSpinner.setText("");
                // 再表示
                onSelectRoleSpinnerClicked(view);
            }
        });
        dialog.setNegativeButton(getText(R.string.cancel), new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int value) {
                // 選択前の状態に戻す
                roleSpinner.setText(buttonText);
            }
        });
        dialog.show();
    }

    @OnClick(R.id.member_registration_button)
    void onRegistrationMemberClicked() {
        String name = nameEditText.getText().toString();
        if (TextUtils.isEmpty(name)) {
            textInputLayout.setErrorEnabled(true);
            textInputLayout.setError(getText(R.string.error_empty_name));
        } else {
            saveItem();
            afterBelong = belongSpinner.getText().toString().split(",");
            changeBelongNo();
            Toast.makeText(this, getText(R.string.member) + " \"" + name + "\" " + getText(R.string.registered), Toast.LENGTH_SHORT).show();
            finish();
        }
    }

    protected void updateItem(int listId) {
        sexButton = (RadioButton) findViewById(sexGroup.getCheckedRadioButtonId());
        String name = nameEditText.getText().toString();
        String name_read = readEditText.getText().toString();
        String sex = (String) sexButton.getText();
        int age = getValue(ageEditText);
        int grade = getValue(gradeEditText);
        String belong = belongConvertToNo();
        String role = roleSpinner.getText().toString();

        dbAdapter.updateMember(listId, name, sex, age, grade, belong, role, name_read);
        FragmentMember.loadName();
    }

    public void update(final int listId) {
        final Button updateBt = (Button) findViewById(R.id.member_registration_button);
        updateBt.setText(R.string.update);
        updateBt.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = nameEditText.getText().toString();
                if (TextUtils.isEmpty(name)) {
                    textInputLayout.setErrorEnabled(true);
                    textInputLayout.setError(getText(R.string.error_empty_name));
                } else {
                    updateItem(listId);
                    afterBelong = belongSpinner.getText().toString().split(",");
                    changeBelongNo();
                    Toast.makeText(getApplicationContext(), getText(R.string.member) + " \"" + name + "\" " + getText(R.string.updated), Toast.LENGTH_SHORT).show();
                    finish();
                }
            }
        });
    }

    protected void saveItem() {
        sexButton = (RadioButton) findViewById(sexGroup.getCheckedRadioButtonId());

        String name = nameEditText.getText().toString();
        String name_read = readEditText.getText().toString();
        String sex = (String) sexButton.getText();
        int age = getValue(ageEditText);
        int grade = getValue(gradeEditText);
        String belong = belongConvertToNo();
        String role = roleSpinner.getText().toString();

        dbAdapter.open();
        dbAdapter.saveName(name, sex, age, grade, belong, role, name_read);
        dbAdapter.close();
    }

    public static int getValue(EditText totext) {
        String text = totext.getText().toString();
        int a = 0;
        if (text.length() > 0) {
            a = Integer.parseInt(text);
        }
        return a;

    }

    public static String belongConvertToNo() {
        dbAdapter.open();
        String belongText = belongSpinner.getText().toString();
        String[] belongTextArray = belongText.split(",");
        StringBuilder BelongNo = new StringBuilder();

        for (int i = 0; i < belongTextArray.length; i++) {
            String belongGroup = belongTextArray[i];
            for (int j = 0; j < FragmentGroup.ListCount; j++) {
                GroupListAdapter.Group listItem = FragmentGroup.nameList.get(j);
                String groupName = listItem.getGroup();
                if (belongGroup.equals(groupName)) {
                    String listName = String.valueOf(listItem.getId());
                    BelongNo.append(listName + ",");
                }
            }
        }
        dbAdapter.close();

        return BelongNo.toString();
    }

    public void changeBelongNo() {
        int i, j,k,id,belongNo,beforeBelongNo,afterBelongNo;
        String name,name_read="ￚ no data ￚ";
        Boolean change=true;
        GroupListAdapter GPdbAdapter=new GroupListAdapter(this);

        if(beforeBelong!=null){
            beforeBelongNo= beforeBelong.length;
        }else{
            beforeBelongNo=0;
        }

        if(afterBelong!=null){
            afterBelongNo= afterBelong.length;
        }else{
            afterBelongNo=0;
        }

        for (i = 0; i < beforeBelongNo; i++) {
            for (j = 0; j < afterBelongNo; j++) {

                if(beforeBelong[i].equals(afterBelong[j])){
                    change=false;
                }
            }
            if(change){
                GPdbAdapter.open();
                for (k = 0; k < FragmentGroup.ListCount; k++) {
                    GroupListAdapter.Group listItem = FragmentGroup.nameList.get(k);
                    String groupName = listItem.getGroup();
                    if (beforeBelong[i].equals(groupName)) {
                        id = listItem.getId();
                        name=listItem.getGroup();
                        belongNo=listItem.getBelongNo()-1;
                        GPdbAdapter.updateGroup(id,name,name_read,belongNo);
                    }
                }
                GPdbAdapter.close();
            }
        }

        change=true;

        for (i = 0; i < afterBelongNo; i++) {
            for (j = 0; j < beforeBelongNo; j++) {
                if(afterBelong[i].equals(beforeBelong[j])){
                    change=false;
                }
            }
            if(change){
                GPdbAdapter.open();
                for (k = 0; k < FragmentGroup.ListCount; k++) {
                    GroupListAdapter.Group listItem = FragmentGroup.nameList.get(k);
                    String groupName = listItem.getGroup();
                    if (afterBelong[i].equals(groupName)) {
                        id = listItem.getId();
                        name=listItem.getGroup();
                        belongNo=listItem.getBelongNo()+1;
                        GPdbAdapter.updateGroup(id,name,name_read,belongNo);
                    }
                }
                GPdbAdapter.close();
            }
        }
    }

}

File: app/src/main/java/com/pandatone/kumiwake/member/AddMember.kt
package com.pandatone.kumiwake.member

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextUtils
import android.text.TextWatcher
import android.view.KeyEvent
import android.view.MotionEvent
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.*
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatEditText
import androidx.core.view.GestureDetectorCompat
import com.google.android.material.textfield.TextInputEditText
import com.google.android.material.textfield.TextInputLayout
import com.google.firebase.analytics.FirebaseAnalytics
import com.pandatone.kumiwake.*
import com.pandatone.kumiwake.adapter.GroupAdapter
import com.pandatone.kumiwake.adapter.GroupFragmentViewAdapter
import com.pandatone.kumiwake.adapter.MemberAdapter
import com.pandatone.kumiwake.kumiwake.NormalMode
import com.pandatone.kumiwake.member.function.Group
import com.pandatone.kumiwake.member.function.Member
import com.pandatone.kumiwake.member.function.MemberClick
import com.pandatone.kumiwake.member.function.MemberMethods
import com.pandatone.kumiwake.member.members.FragmentGroupMain
import com.pandatone.kumiwake.member.members.FragmentMemberMain
import com.pandatone.kumiwake.others.SelectMember
import com.pandatone.kumiwake.ui.dialogs.DialogWarehouse
import kotlinx.android.synthetic.main.add_member.*


/**
 * Created by atsushi_2 on 2016/02/24.
 */
class AddMember : AppCompatActivity() {
    private var beforeBelong: Array<String>? = null
    private var afterBelong: Array<String>? = null
    private var nameEditText: AppCompatEditText? = null
    private var readEditText: AppCompatEditText? = null
    private var textInputLayout: TextInputLayout? = null
    private var sexGroup: RadioGroup? = null
    private var sexButton: RadioButton? = null
    private var ageEditText: EditText? = null
    private var belongDropdown: AutoCompleteTextView? = null
    private var mbAdapter: MemberAdapter? = null
    private var fromMode: String? = "member"
    private val groupList: ArrayList<Group>
        get() {
            return GroupAdapter(this).getAllGroups()
        }
    private var dialogShown = false
    private lateinit var mDetector: GestureDetectorCompat

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        FirebaseAnalyticsEvents.firebaseAnalytics = FirebaseAnalytics.getInstance(this)
        setTheme(StatusHolder.nowTheme)
        setContentView(R.layout.add_member)
        mbAdapter = MemberAdapter(this)
        FragmentGroupMain.gpAdapter = GroupAdapter(this)
        FragmentGroupMain.listAdp = GroupFragmentViewAdapter(this, groupList)
        beforeBelong = null
        afterBelong = null
        findViews()
        ageEditText!!.setText("")
        val i = intent
        val member = i.getSerializableExtra(AddMemberKeys.MEMBER.key) as Member?
        fromMode = i.getStringExtra(AddMemberKeys.FROM_MODE.key)
        val memberImg = findViewById<ImageView>(R.id.memberIcon)
        sexGroup!!.setOnCheckedChangeListener { _, checkedId: Int ->
            when (checkedId) {
                R.id.manBtn -> memberImg.setColorFilter(PublicMethods.getColor(this, R.color.man))
                R.id.womanBtn -> memberImg.setColorFilter(PublicMethods.getColor(this, R.color.woman))
            }
        }
        if (member != null) {
            switch_continuously_mode.visibility = View.GONE
            setItem(member)
            member_registration_continue_btn.visibility = View.GONE
        }

        var yomigana = ""

        nameEditText?.addTextChangedListener(object : TextWatcher {

            override fun afterTextChanged(s: Editable) {
                if (s.matches("^[a-zA-Z0-9ぁ-ん\\s]+$".toRegex()) || s.toString() == "") {
                    yomigana = s.toString()
                }
                readEditText?.setText(yomigana)
            }

            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
        })

        Toast.makeText(this, getText(R.string.double_tap), Toast.LENGTH_SHORT).show()
        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        mDetector = GestureDetectorCompat(this, MyGestureListener(imm, nameEditText as EditText))
        mDetector.setOnDoubleTapListener(MyGestureListener(imm, nameEditText as EditText))
    }

    //スクロールビューの場合こっち呼ぶ
    override fun dispatchTouchEvent(event: MotionEvent?): Boolean {
        super.dispatchTouchEvent(event)
        mDetector.onTouchEvent(event)
        return super.onTouchEvent(event)
    }

    //Viewの宣言
    private fun findViews() {
        nameEditText = findViewById<View>(R.id.input_name) as TextInputEditText
        readEditText = findViewById<View>(R.id.input_name_read) as TextInputEditText
        textInputLayout = findViewById<View>(R.id.member_form_input_layout) as TextInputLayout
        sexGroup = findViewById<View>(R.id.sexGroup) as RadioGroup
        ageEditText = findViewById<View>(R.id.input_age) as EditText
        belongDropdown = findViewById<View>(R.id.select_group_choicer) as AutoCompleteTextView
        belongDropdown?.let { button -> button.setOnClickListener { onSelectGroupDropdownClicked(button) } }
        findViewById<Button>(R.id.member_registration_finish_btn).setOnClickListener { register(true) }
        findViewById<Button>(R.id.member_registration_continue_btn).setOnClickListener { register(false) }
        findViewById<Button>(R.id.member_cancel_btn).setOnClickListener { finish() }
        findViewById<Button>(R.id.switch_continuously_mode).setOnClickListener {
            val intent = Intent(this, AddMemberInBulk::class.java)
            intent.putExtra(AddMemberKeys.FROM_MODE.key, fromMode)
            finish()
            startActivity(intent)
        }
    }

    //Updateの場合の初期アイテム表示
    private fun setItem(member: Member) {
        val listId = member.id
        val name = member.name
        val read = member.read
        val sex = member.sex
        val age = member.age.toString()
        val belong = MemberClick.viewBelong(member, FragmentGroupMain.groupList)

        nameEditText!!.setText(name)
        readEditText!!.setText(read)
        if (sex == getString(R.string.woman)) {
            sexGroup!!.check(R.id.womanBtn)
        } else {
            sexGroup!!.check(R.id.manBtn)
        }
        ageEditText!!.setText(age)
        belongDropdown!!.setText(belong)

        update(listId)

    }


    @SuppressLint("SetTextI18n")
    internal fun onSelectGroupDropdownClicked(view: View) {
        if (!dialogShown) { //二回呼ばれる問題の解決
            dialogShown = true
            // 選択中の候補を取得
            val buttonText = belongDropdown!!.text.toString()
            val textArray = buttonText.split(", ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            // 候補リスト
            val list = ArrayList<String>()
            for (group in groupList) {
                list.add(group.name)
            }
            val belongArray = list.toTypedArray()
            // 選択リスト
            val checkArray = BooleanArray(belongArray.size)
            for (i in belongArray.indices) {
                checkArray[i] = false
                for (data in textArray) {
                    if (belongArray[i] == data) {
                        checkArray[i] = true
                        break
                    }
                }
            }
            // ダイアログを生成
            val dialog = AlertDialog.Builder(this)

            // 選択イベント
            dialog.setMultiChoiceItems(belongArray, checkArray) { _, _, _ -> }
            dialog.setPositiveButton(getText(R.string.decide)) { _, _ ->
                val buffer = StringBuilder()
                for (i in belongArray.indices) {
                    if (checkArray[i]) {
                        buffer.append((if (buffer.isEmpty()) "" else ", ") + belongArray[i])
                    }
                }
                belongDropdown!!.setText(buffer)
                dialogShown = false
            }
            dialog.setNeutralButton(getText(R.string.clear)) { _, _ ->
                belongDropdown!!.setText("")
                dialogShown = false
                // 再表示
                onSelectGroupDropdownClicked(view)
            }
            dialog.setNegativeButton(getText(R.string.cancel)) { _, _ -> dialogShown = false }
            dialog.show()

            beforeBelong = textArray
        }
    }

    //バックキーの処理
    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            DialogWarehouse(supportFragmentManager).decisionDialog("KUMIWAKE", getString(R.string.edit_exit_confirmation)) { finish() }
            return true
        }
        return false
    }

    //メンバー登録　finish true:終了, false:続けて登録
    private fun register(finish: Boolean) {
        val name = nameEditText!!.text!!.toString()
        if (TextUtils.isEmpty(name)) {
            textInputLayout!!.isErrorEnabled = true
            textInputLayout!!.error = getText(R.string.error_empty_name)
        } else {
            saveItem()
            afterBelong = belongDropdown!!.text.toString().split(", ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            MemberMethods.updateBelongNo(this)
            val newMember = MemberAdapter(this).newMember

            when (fromMode) {
                "normal" -> NormalMode.memberArray.add(newMember)
                "others" -> SelectMember.memberArray.add(newMember)
            }

            Toast.makeText(this, getText(R.string.member).toString() + " \"" + name + "\" " + getText(R.string.registered), Toast.LENGTH_SHORT).show()
            FirebaseAnalyticsEvents.memberRegisterEvent(newMember)

            if (finish) {
                finish()
            } else {
                val intent = intent
                finish()
                startActivity(intent)
            }
        }
    }

    //メンバー登録処理
    private fun saveItem() {
        sexButton = findViewById<View>(sexGroup!!.checkedRadioButtonId) as RadioButton

        val name = nameEditText!!.text!!.toString()
        val read = readEditText!!.text!!.toString()
        val sex = sexButton!!.text as String
        val age = getValue(ageEditText!!)
        val belong = MemberMethods.belongConvertToNo(belongDropdown!!.text.toString(), groupList)

        mbAdapter!!.saveName(name, sex, age, belong, read)
    }

    //メンバー情報更新ボタンリスナー
    private fun update(listId: Int) {
        val updateBt = findViewById<View>(R.id.member_registration_finish_btn) as Button
        updateBt.setText(R.string.update)
        updateBt.setOnClickListener {
            val name = nameEditText!!.text!!.toString()
            if (TextUtils.isEmpty(name)) {
                textInputLayout!!.isErrorEnabled = true
                textInputLayout!!.error = getText(R.string.error_empty_name)
            } else {
                updateItem(listId)
                afterBelong = belongDropdown!!.text.toString().split(", ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                MemberMethods.updateBelongNo(this)
                Toast.makeText(applicationContext, getText(R.string.member).toString() + " \"" + name + "\" " + getText(R.string.updated), Toast.LENGTH_SHORT).show()
                finish()
            }
        }
    }

    //メンバー情報更新処理
    private fun updateItem(listId: Int) {
        sexButton = findViewById<View>(sexGroup!!.checkedRadioButtonId) as RadioButton
        val name = nameEditText!!.text!!.toString()
        val read = readEditText!!.text!!.toString()
        val sex = sexButton!!.text as String
        val age = getValue(ageEditText!!)
        val belong = MemberMethods.belongConvertToNo(belongDropdown!!.text.toString(), groupList)

        mbAdapter!!.updateMember(listId, name, sex, age, belong, read)
        FragmentMemberMain().loadName()
    }

    //String to Int
    private fun getValue(toText: EditText): Int {
        val text = toText.text.toString()
        var a = 0
        if (text.isNotEmpty()) {
            a = Integer.parseInt(text)
        }
        return a
    }

}







